### DISCUSSION TITLE
Do you feel the need of a better VSCode extension?

### DISCUSSION THREAD

**Discussion by giacomorebonato** [json5/json5#333] [Q&A]
The main official extension is unmaintained and in many cases [it breaks syntax highlight](https://github.com/mrmlnc/vscode-json5/issues/4).

Hence my way to use JSON5 in VSCode is to:

1. associate `.json5` extension to JavaScript

```json
  "files.associations": {
    "*.json5": "javascript"
  },
```

2. and then to disable JavaScript validation to not see the following error:

<img width="393" alt="test-json5" src="https://github.com/microsoft/vscode/assets/630796/8fb68288-1755-44d1-a534-11529f60bea7">  

**Solution:**

```json
"javascript.validate.enable": false
```

## Pros

I can now use eslint to lint my JSON5 files using [eslint-plugin-jsonc](https://github.com/ota-meshi/eslint-plugin-jsonc)

## Cons 

I have disabled JavaScript validation for any files associated with the language.

## Conclusion

May I ask your thoughts regarding this scenario? 
Are you using the JSON5 extension or are you building like me some custom tooling?

## Other discussions

* https://github.com/microsoft/vscode/issues/190678
* https://github.com/ota-meshi/eslint-plugin-jsonc/issues/256

**Comment by jasonswearingen** [comment]
there now **IS** a better vscode extension!  https://marketplace.visualstudio.com/items?itemName=blueglassblock.better-json5

I installed it and it's working pretty great!

**Comment by Pomax** [comment]
The world definitely needs an extension that actually lives under the `json5` org here in github, so folks can trust it. As nice as "better json5" might be, with only 1k installs compared to the (far less useful) "json5 syntax" extension, people are going to ignore it and keep installing the other one.

Having "json5" as publisher, rather than "mrmlnc" or "BlueGlassBlock" makes a huge difference in this case.

(on a practical note, better-json5 doesn't have the TS/Prettier equivalent of the `ignore` statement, so that single lines/blocks can be marked as "there is a _very_ good reason for this specific thing to violate the general formatting rules, do not interfere with it")
